File: 백준/Silver/1931. 회의실 배정/회의실 배정.py
N = int(input())
data = []
for _ in range(N):
    s, e = map(int, input().split())
    data.append((s, e))

data.sort(reverse=True)

cnt = 0
# 뒤에서부터 시작 시간을 찾음
# 시작 시간이 가장 느린 것
start = 2**31
for s, e in data:
    # 시작 시간이 느리고,
    # 이전 시작 시간보다 종료 시간이 빠르면 추가
    if start >= e:
        start = s
        cnt += 1

print(cnt)

File: 백준/Gold/1931. 회의실 배정/회의실 배정.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split('\n');

let n = Number(input[0]);
let arr = [];

for (let i = 1; i <= n; i++) {
  let time = input[i].split(' ').map(Number);
  arr.push(time);
}

arr.sort((x, y) => {
  if (x[1] === y[1]) return x[0] - y[0]; // 같으면, 시작 시점
  // 종료 시점 기준 정렬
  return x[1] - y[1];
});

let cnt = 0;
let lastTime = 0;

for ([s, e] of arr) {
  if (lastTime <= s) {
    lastTime = e;
    cnt += 1;
  }
}

console.log(cnt);
